crates/bls-crypto/src/bls/signature.rs

```diff
@@ -66,6 +66,24 @@ impl Signature {
             .into()
     }
 
+    /// Computes the dot product of a vector of signatures and a vector of exponents.
+    pub fn batch(r: &[Fr], signatures: &[&Signature]) -> Signature {
+        // Get everything in the right types
+        let projective_elements = signatures
+            .iter()
+            .map(|s| s.as_ref().clone())
```

Comment:
How about leaving this to the caller? This can be misleading for the caller to assume that they passed a reference and then see extra memory usage